components/tikv_util/src/metrics/mod.rs

```diff
@@ -115,6 +186,100 @@ pub fn convert_record_pairs(m: HashMap<String, u64>) -> RecordPairVec {
         .collect()
 }
 
+/// MetricsCompactLevel defines the level of compact metrics sample data, a higher level of
+/// means smaller data size and more information loss.
+#[derive(Serialize_repr, Deserialize_repr, Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(u32)]
+pub enum MetricsCompactLevel {
```

Comment:
MetricCompactPolicy might be a better choice ? For instance:
pub enum MetricCompactPolicy{
    NoCompaction = 0,   
    LoseLessCompaction = 1, 
    LossyCompaction = 2,
}